invoices/invoiceregistry.go

```diff
@@ -465,19 +469,67 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 			return nil, errNoUpdate
 		}
 
-		// If the invoice is already canceled, there is no further
-		// checking to do.
-		if inv.Terms.State == channeldb.ContractCanceled {
-			debugLog("invoice already canceled")
+		// Only accept payments to open invoices.
+		if inv.Terms.State != channeldb.ContractOpen {
+			debugLog(fmt.Sprintf(
+				"invoice in state %v and no longer open",
+				inv.Terms.State))
 			return nil, errNoUpdate
 		}
 
-		// If an invoice amount is specified, check that enough
-		// is paid. Also check this for duplicate payments if
-		// the invoice is already settled or accepted.
-		if inv.Terms.Value > 0 && amtPaid < inv.Terms.Value {
-			debugLog("amount too low")
-			return nil, errNoUpdate
+		// Start building the accept descriptor.
+		acceptDesc := &channeldb.HtlcAcceptDesc{
+			Amt:          amtPaid,
+			Expiry:       expiry,
+			AcceptHeight: currentHeight,
+		}
+
+		// If an invoice amount is specified and this is not an mpp,
+		// check that the exact amount is paid.
+		var canSettle bool
+		if mpp == nil {
+			if inv.Terms.Value > 0 && amtPaid < inv.Terms.Value {
+				debugLog("amount too low and no mpp")
+				return nil, errNoUpdate
+			}
+			canSettle = true
+		} else {
+			// Check whether total amt matches other htlcs in the
+			// set.
+			for _, htlc := range inv.Htlcs {
+				// Only consider accepted mpp htlcs.
+				if htlc.State != channeldb.HtlcStateAccepted {
+					continue
+				}
+
+				if htlc.TotalAmt == 0 {
```

Comment:
when would this happen? why would we have accepted it?

Reply:
I don't remember exactly, but I think this was for non-mpp htlcs. But if they would be present for the hodl invoice case, we can never get here. Removed.